src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -373,26 +379,24 @@ public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key,
             algorithm = Pem.DEFAULT_ALGO;
         }
 
+        char[] passwd = password.clone();
         try {
-            SecretKeyFactory factory;
-            if (provider == null) {
-                factory = SecretKeyFactory.getInstance(algorithm);
-            } else {
-                factory = SecretKeyFactory.getInstance(algorithm, provider);
-            }
-            skey = factory.generateSecret(keySpec);
-        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
-            throw new IllegalArgumentException(e);
+            return encryptKeyImpl(getEncoding(de), algorithm,
+                generateSecretKey(password, algorithm, provider), params,
+                provider, null);
+        } finally {
+            Arrays.fill(passwd, (char)0x0);
         }
-        return encryptKeyImpl(key, algorithm, skey, params, provider, null);
     }
 
     /**
      * Creates and encrypts an {@code EncryptedPrivateKeyInfo} from a given
-     * {@code PrivateKey} and password.  Default algorithm and parameters are
+     * {@code DEREncodable} and password.  Default algorithm and parameters are
      * used.
      *
-     * @param key the {@code PrivateKey} to be encrypted
+     * @param de the {@code DEREncodable} to be encrypted. Usage
+     *            limited to {@link PrivateKey}, {@link KeyPair}, and
+     *            {@link PKCS8EncodedKeySpec}.
      * @param password the password used in the PBE encryption.  This array
      *                 will be cloned before being used.
      * @return an {@code EncryptedPrivateKeyInfo}
```

Comment:
Next line: 

"@throws IllegalArgumentException on initialization errors based on the
arguments passed to the method"

What does this mean - you need to be more specific about what are the error cases. Otherwise it will not be helpful for TCK test writers on how to test this.